### PR TITLE
Tila 384 allocation by basket

### PR DESCRIPTION
**Pull request by aprenfors**
There are no functional changes here, just refactored django models out of allocation models for less tight coupling and created an intermediate builder to handle initializing AllocationData

Also refactored test cases to use assertpy.

### CODE REVIEW

File: reservations/allocation_data_builder.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                        occurrences=application_event.get_all_occurrences(),
+                        period_start=self.period_start,
+                        period_end=self.period_end,
+                        baskets=event_baskets.get(application_event.id),
+                        space_ids=space_ids,
+                        begin=application_event.begin,
+                        end=application_event.end,
+                        min_duration=application_event.min_duration,
+                        max_duration=application_event.max_duration
+                        if application_event.max_duration is not None
```
Comment (nicobav) [inline reservations/allocation_data_builder.py:54]: do we need explicit `None` check here? 
For example using  `if application_event.max_duration` to also check against zero (0) value => so what if the max_duration is zero, should it be count or not as max_duration?

File: reservations/tests/test_allocation_model.py [hunk 2]
```diff
@@ -115,11 +123,28 @@ def test_mapping_application_round_baskets(
...
+        application_round=application_round_with_reservation_units
+    ).get_allocation_data()
+
+    assert_that(data.baskets).contains_key(
+        application_round_basket_one.id, application_round_basket_two.id
+    )
+
+    assert_that(data.baskets[application_round_basket_one.id]).has_id(
+        application_round_basket_one.id
+    ).has_order_number(
```
Comment (nicobav) [inline reservations/tests/test_allocation_model.py:136]: this dynamic assertion is nice :sunglasses:

**Review by nicobav — APPROVED** [review]

File: reservations/allocation_data_builder.py (hunk 1 above)
Comment (aprenfors) [reply reservations/allocation_data_builder.py:54]: There is validation that checks that max duration is none or larger than min duration, so there should be no need to check for zeros.

File: reservations/tests/test_allocation_model.py (hunk 2 above)
Comment (aprenfors) [reply reservations/tests/test_allocation_model.py:136]: Yeah, i kind of find magic like this a bit disturbing, could be hard to debug, but seems to work.

**Pull request merged**